### PR TITLE
Add comment for behavior tree

### PR DESCRIPTION
**Pull request by DevilSky93**

### CODE REVIEW

**Comment by samidalil** [comment]
[Documentation du module BehaviorTree](https://trello.com/c/fhRaif0U/11-documentation-du-module-behaviortree)

File: Assets/General/BehaviorTree/BehaviorTreeAgent.cs [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
+        /// <summary>
+        /// Setup the tree by adding node (selector, sequences ...)
+        /// </summary>
         private void Start()
         {
             _root = SetupTree();
         }
 
+        /// <summary>
+        /// Evaluate every node from the tree by browsing every children
```
Comment (Neffros) [inline Assets/General/BehaviorTree/BehaviorTreeAgent.cs:24]: Consider precising that this evaluation is each frame.

File: Assets/General/BehaviorTree/Node.cs [hunk 2]
```diff
@@ -2,41 +2,81 @@
...
+        /// <summary>
+        /// Create node with no parent
+        /// </summary>
+        protected Node()
         {
             Parent = null;
         }
 
+        /// <summary>
+        /// Attach child node with this current node
```
Comment (Neffros) [inline Assets/General/BehaviorTree/Node.cs:48]: Consider this phrasing: "Attach child node to this instance of Node"

File: Assets/General/BehaviorTree/Node.cs [hunk 3]
```diff
@@ -2,41 +2,81 @@
...
+        /// Evaluate children node
+        /// </summary>
+        /// <returns>Return FAILURE by default</returns>
         public virtual NodeState Evaluate() => NodeState.FAILURE;
 
+        /// <summary>
+        /// Add data to the dictionary
+        /// </summary>
+        /// <param name="key">Key to be add</param>
+        /// <param name="value">Value to be add</param>
```
Comment (Neffros) [inline Assets/General/BehaviorTree/Node.cs:69]: Grammar mistake: "Value to be added" instead

File: Assets/General/BehaviorTree/Node.cs [hunk 4]
```diff
@@ -2,41 +2,81 @@
...
         public void SetData(string key, object value)
         {
             _dataContext[key] = value;
         }
 
+        /// <summary>
+        /// Add data to the dictionary of the root node
```
Comment (Neffros) [inline Assets/General/BehaviorTree/Node.cs:76]: Grammar mistake: "Adds data to the root node's dictionary"

File: Assets/General/BehaviorTree/Node.cs [hunk 5]
```diff
@@ -2,41 +2,81 @@
...
         }
 
+        /// <summary>
+        /// Evaluate children node
+        /// </summary>
+        /// <returns>Return FAILURE by default</returns>
         public virtual NodeState Evaluate() => NodeState.FAILURE;
 
+        /// <summary>
+        /// Add data to the dictionary
```
Comment (samidalil) [inline Assets/General/BehaviorTree/Node.cs:66]: Typo: "Adds"

**Review by samidalil — CHANGES_REQUESTED** [review]

**Review by samidalil — APPROVED** [review]

**Pull request merged**